File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/dtos/ServiceRequest.java
package dev.jefy.connectpro.portfolio.applicaion.dtos;


import java.util.List;
import java.util.UUID;

import dev.jefy.connectpro.shared.application.dtos.PricingData;

/**
 * @author Jôph Yamba
 */
public record ServiceRequest(
    UUID portfolioId,
    String title, 
    String description,
    UUID categoryId, 
    List<String> tags, 
    PricingData pricing
) {}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/dtos/ServiceRequest.kt
package dev.jefy.connectpro.portfolio.application.dtos

import java.util.*

data class ServiceRequest(
    val portfolioId: UUID,
    val name: String,
    val description: String,
    val categoryId: UUID,
    val tags: Set<String>,
){
    init {
        require(name.isNotBlank()) { "title must not be blank" }
        require(description.isNotBlank()) { "description must not be blank" }
    }
}